### PR TITLE
Jd/sparsify with tolerance

### LINKED ISSUE #9: Implement Sparsification approach to PTDF

### PR DESCRIPTION
**Pull request by jd-lara**
Thanks for opening a PR to PowerNetworkMatrices.jl, please take note of the following when making a PR:

Check the [contributor guidelines](https://nrel-siip.github.io/PowerNetworkMatrices.jl/stable/code_base_developer_guide/developer/)

closes #9

### CODE REVIEW

File: src/common.jl [hunk 1]
```diff
@@ -124,10 +124,150 @@ function calculate_BA_matrix(
...
 
+function calculate_BA_matrix_full(
+    branches,
+    bus_lookup::Dict{Int, Int})
+    BA_I = Int[]
+    BA_J = Int[]
+    BA_V = Float64[]
+
+    for (ix, b) in enumerate(branches)
+        if isa(b, PSY.DCBranch)
```
Comment (jd-lara) [inline src/common.jl:135]: @alefcastelli delete this check. By calling the PNM function `get_ac_branches` when passing the arguments to the function is should be addressed.

File: src/ptdf_calculations.jl [hunk 2]
```diff
@@ -3,19 +3,35 @@ Power Transfer Distribution Factors (PTDF) indicate the incremental change in re
...
+    return PTDF(new_mat, mat.axes, mat.lookup, mat.subnetworks, Ref(tol))
 end
 
 function _buildptdf(
     branches,
     nodes::Vector{PSY.Bus},
     bus_lookup::Dict{Int, Int},
     dist_slack::Vector{Float64},
-    linear_solver::String = "Dense")
+    linear_solver::String = "KLU")
```
Comment (jd-lara) [inline src/ptdf_calculations.jl:34]: remove default in this function

**Review by rodrigomha — APPROVED** [review]
LGTM

File: src/ptdf_calculations.jl (hunk 2 above)
Comment (alefcastelli) [reply src/ptdf_calculations.jl:34]: done

File: src/common.jl (hunk 1 above)
Comment (alefcastelli) [reply src/common.jl:135]: that is part of a function that has been removed in the latest commit

**Pull request merged**